lib/seek/roles/project_related_roles.rb

```diff
@@ -157,7 +157,7 @@ def is_in_any_gatekept_projects?
         # any for projects that have been removed, and resolving the mask
         def resolve_admin_defined_role_projects
           admin_defined_role_projects.each do |role|
-            role.destroy unless projects.include?(role.project)
+            role.destroy unless group_memberships.collect(&:project).include?(role.project)
```

Comment:
Is it done this way because of the issue where the association doesn't work on unsaved associations?

Reply:
yes, otherwise you can't add somebody to a project and assign a role without saving in between. At first I updated the test to do an intermediate save, but changed my mind and decided this was safer ( https://github.com/seek4science/seek/pull/929/commits/a8a92d22d1d7101aebffe7956781cf23e7ab0546 )